File: 303_RangeSumQueryImmutable/rangeSumQueryImmutable.js
/**
 * @param {number[]} nums
 */
var NumArray = function(nums) {
    let prefixSums = new Array(nums.length).fill(nums[0])
    for(let i = 1; i < nums.length; i++) {
    	prefixSums[i] = prefixSums[i - 1] + nums[i]
    }
};

/** 
 * @param {number} left 
 * @param {number} right
 * @return {number}
 */
NumArray.prototype.sumRange = function(left, right) {
    if (left === 0) {
    	return NumArray[right]
    }
    return NumArray[right] - NumArray[left - 1]
};

/** 
 * Your NumArray object will be instantiated and called as such:
 * var obj = new NumArray(nums)
 * var param_1 = obj.sumRange(left,right)
 */

File: 303_RangeSumQueryImmutable/rangeSumQueryImmutable.py
class NumArray:

    def __init__(self, nums: List[int]):
        self.prefixSumArr = [0] * len(nums)
        self.prefixSumArr[0] = nums[0]
        for i in range(1, len(nums)):
            self.prefixSumArr[i] = nums[i] + self.prefixSumArr[i - 1]

    def sumRange(self, left: int, right: int) -> int:
        if left == 0:
            return self.prefixSumArr[right]
        return self.prefixSumArr[right] - self.prefixSumArr[left - 1]


# Your NumArray object will be instantiated and called as such:
# obj = NumArray(nums)
# param_1 = obj.sumRange(left,right)